### PR TITLE
CB-10552: Pull all the plugin docs to Cordova

### PR DESCRIPTION
**Pull request by sarangan12**
Pulling plugin docs to cordova

@dblotsky Can you take a look at this PR?

### CODE REVIEW

**Comment by dblotsky** [comment]
@stevengill @purplecabbage @csantanapr please take a look.

**Comment by dblotsky** [comment]
@sarangan12 looks like the mailing list is quiet about this. @riknoll any feedback?

**Comment by stevengill** [comment]
I'm testing this now. Just pulled it into my local copy

**Comment by stevengill** [comment]
BTW, I had a merge conflict when pulling this in. Need to rebase this off the latest master I believe.

**Comment by purplecabbage** [comment]
Looks good!

**Comment by stevengill** [comment]
Just finished my run through. 

LGTM.

It is nice to not leave the site to see the docs. I also like how it pulls the latest release and warns users about the version. More useful than npm's readme.

File: tools/bin/fetch_docs.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+function generateFrontMatter (useDirectSrcURI, filePath, name, version) {
+    var obj                 = {};
+
+    if(useDirectSrcURI) {
+        filePath            = filePath.replace("https://raw.githubusercontent.com/", "");
+        var filePathSplit   = filePath.split("/");
+        name                = filePathSplit[0] + "/" + filePathSplit[1];
+        version             = filePathSplit[2];
+        filePath            = filePathSplit.slice(3).join("/");
+    }
```
Comment (dblotsky) [inline tools/bin/fetch_docs.js]: Please use a regex here.

File: www/_layouts/docs.html [hunk 2]
```diff
@@ -183,6 +183,21 @@ <h2 class="site-toc-title">{{ page.toc_text }}</h2>
...
+                <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                {{ page.note }}
+                <a href="{{ page.note_link }}">
+                    {{ page.note_link_text }}
+                </a>
+            </div>
+
+        {% endif %}
```
Comment (dblotsky) [inline www/_layouts/docs.html]: Please adjust this code with respect to the comment on `fetch_docs.js`, and to trigger when `plugin_version` is specified.

File: tools/bin/fetch_docs.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    obj.edit_link           = getRepoURIForFrontMatter(name, version, filePath);
+    obj.note_link           = "https://github.com/" + name + "/releases";
+    obj.note                = "This documentation is for the " + version + " version of the " + name + ".";
+    obj.note_link_text      = "Visit Github for other versions.";
```
Comment (dblotsky) [inline tools/bin/fetch_docs.js]: Please move this logic to the Jekyll side. Instead set a variable called something like `plugin_version`, and construct the link in HTML.

File: tools/bin/fetch_docs.js (hunk 1 above)
Comment (sarangan12) [inline tools/bin/fetch_docs.js]: Done

File: www/_layouts/docs.html (hunk 2 above)
Comment (sarangan12) [inline www/_layouts/docs.html]: Done

File: tools/bin/fetch_docs.js (hunk 3 above)
Comment (sarangan12) [inline tools/bin/fetch_docs.js]: Done

**Comment by sarangan12** [comment]
@dblotsky Can you review and merge this PR?

**Comment by csantanapr** [comment]
Sorry I'm late to this one, It looks like @stevengill already review.

Just some dumb questions
1. What impact this has to translation process?
2. Everything automated to avoid maintenance headache to keep website in sync?

**Pull request closed without merging**